src/Rules/Functions/PrintfParametersRule.php

```diff
@@ -109,6 +112,17 @@ public function processNode(Node $node, Scope $scope): array
 
 	private function getPlaceholdersCount(string $functionName, string $format): int
 	{
+		try {
+			@sprintf($format);
+		} catch (ArgumentCountError $e) {
+			sscanf($e->getMessage(), '%i arguments are required, %i given', $required, $given);
+
+			if (is_int($required)) {
+				return $required - 1;
+			}
+		}
+
+		// fallback for php < 8
```

Comment:
the whole regex parsing can be deleted when phpstan min-version is raised to php8+